daemon/cluster/convert/container.go

```diff
@@ -211,7 +214,10 @@ func configReferencesFromGRPC(sr []*swarmapi.ConfigReference) []*types.ConfigRef
 	return refs
 }
 
-func containerToGRPC(c types.ContainerSpec) (*swarmapi.ContainerSpec, error) {
+func containerToGRPC(c *types.ContainerSpec) (*swarmapi.ContainerSpec, error) {
+	if c == nil {
+		return nil, nil
+	}
```

Comment:
I have a slight preference for doing this check in the caller, because returning a `nil` error suggests that it's safe to dereference the pointer being returned.